pubsub/subscriptions/subscription_test.go

```diff
@@ -43,38 +51,72 @@ func setup(t *testing.T) *pubsub.Client {
 	ctx := context.Background()
 	tc := testutil.SystemTest(t)
 
-	topicID = "test-sub-topic"
-	subID = "test-sub"
 	var err error
 	client, err := pubsub.NewClient(ctx, tc.ProjectID)
 	if err != nil {
 		t.Fatalf("failed to create client: %v", err)
 	}
 
-	// Cleanup resources from the previous tests.
 	once.Do(func() {
-		topic := client.Topic(topicID)
-		ok, err := topic.Exists(ctx)
-		if err != nil {
-			t.Fatalf("failed to check if topic exists: %v", err)
-		}
-		if ok {
-			if err := topic.Delete(ctx); err != nil {
-				t.Fatalf("failed to cleanup the topic (%q): %v", topicID, err)
+		topicID = fmt.Sprintf("%s-%d", topicPrefix, time.Now().UnixNano())
+		subID = fmt.Sprintf("%s-%d", subPrefix, time.Now().UnixNano())
+
+		// Cleanup resources from the previous tests.
+		it := client.Topics(ctx)
+		for {
+			t, err := it.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				return
+			}
+			tID := t.ID()
+			p := strings.Split(tID, "-")
```

Comment:
possibly SplitN here and in the subscription equivalent in case someone has a `sub-123-456` or something?